magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala

```diff
@@ -336,91 +334,66 @@ class CheckUpdateEventsTask(
-
-    def updateFailed(e: StackEvent): Boolean = {
-      val failed = e.resourceStatusAsString.contains("FAILED") || e.resourceStatusAsString.contains("ROLLBACK")
-      logger.debug(s"${e.resourceStatusAsString} - failed = $failed")
-      failed
-    }
-
-    def fail(reporter: DeployReporter, e: StackEvent): Unit = reporter.fail(
-      s"""${e.logicalResourceId}(${e.resourceType}}: ${e.resourceStatusAsString}
-            |${e.resourceStatusReason}""".stripMargin)
-  }
 
   def description = s"Checking events on update for stack $stackLookupStrategy"
 }
 
-/*
-We're sub-classing `CheckUpdateEventsTask` in order to first check if a ChangeSet has yielded any changes - only poll for CloudWatch Events if it did.
-Ideally this check would sit in `CheckUpdateEventsTask` itself, however `CheckUpdateEventsTask` is used by the `AmiCloudFormationParameter` deployment type, which is not yet ChangeSet aware.
-TODO Make `AmiCloudFormationParameter` deployment type ChangeSet aware.
- */
-class CheckUpdateEventsForChangeSetTask(
-  region: Region,
-  stackLookup: CloudFormationStackMetadata
-)(implicit override val keyRing: KeyRing) extends CheckUpdateEventsTask(region, stackLookup.strategy) {
-  override def execute(resources: DeploymentResources, stopFlag: => Boolean): Unit = {
-    CloudFormation.withCfnClient(keyRing, region, resources) { cfnClient =>
-      val (stackName, _, _) = stackLookup.lookup(resources.reporter, cfnClient)
-      val changeSetName = stackLookup.changeSetName
+class CloudFormationStackEventPoller(stackName: String, cfnClient: CloudFormationClient, resources: DeploymentResources, stopFlag: => Boolean) extends Loggable {
+  private def reportEvent(reporter: DeployReporter, e: StackEvent): Unit = {
+    reporter.info(s"${e.logicalResourceId} (${e.resourceType}): ${e.resourceStatusAsString}")
+    if (e.resourceStatusReason != null) reporter.verbose(e.resourceStatusReason)
+  }
+  private def isStackEvent(stackName: String)(e: StackEvent): Boolean =
+    e.resourceType == "AWS::CloudFormation::Stack" && e.logicalResourceId == stackName
 
-      val describeRequest = DescribeChangeSetRequest.builder().changeSetName(changeSetName).stackName(stackName).build()
-      val describeResponse = cfnClient.describeChangeSet(describeRequest)
+  private def updateStart(stackName: String)(e: StackEvent): Boolean =
+    isStackEvent(stackName)(e) && (e.resourceStatusAsString == "UPDATE_IN_PROGRESS" || e.resourceStatusAsString == "CREATE_IN_PROGRESS")
 
-      if (describeResponse.changes.isEmpty) {
-        resources.reporter.info(s"No changes to perform for $changeSetName on stack $stackName")
-      } else {
-        super.execute(resources, stopFlag)
-      }
-    }
+  private def updateComplete(stackName: String)(e: StackEvent): Boolean =
+    isStackEvent(stackName)(e) && (e.resourceStatusAsString == "UPDATE_COMPLETE" || e.resourceStatusAsString == "CREATE_COMPLETE")
+
+  private def updateFailed(e: StackEvent): Boolean = {
```

Comment:
Minor, but I recommend defaulting to `private[this]` unless you have a good reason to do otherwise.

Reply:
Done, thanks!